src/vs/workbench/contrib/chat/browser/chatEditing/simpleBrowserEditorOverlay.ts

```diff
@@ -0,0 +1,299 @@
+		}, {
+			id: 'element-screenshot-' + Date.now(),
+			name: 'Element Screenshot',
+			fullName: 'Element Screenshot',
+			kind: 'image',
+			value: screenshot.buffer
+		});
+	}
+
+
+	getDisplayNameFromOuterHTML(outerHTML: string): string {
+		const firstElementMatch = outerHTML.match(/^<(\w+)([^>]*?)>/);
+		if (!firstElementMatch) {
+			throw new Error('No outer element found');
+		}
+
+		const tagName = firstElementMatch[1];
+		const idMatch = firstElementMatch[2].match(/\s+id\s*=\s*["']([^"']+)["']/i);
+		const id = idMatch ? `#${idMatch[1]}` : '';
+		const classMatch = firstElementMatch[2].match(/\s+class\s*=\s*["']([^"']+)["']/i);
+		const className = classMatch ? `.${classMatch[1].replace(/\s+/g, '.')}` : '';
+		return `${tagName}${id}${className}`;
+	}
+
+	dispose() {
+		this.hide();
+		this._showStore.dispose();
+	}
+
+	getDomNode(): HTMLElement {
+		return this._domNode;
+	}
+
+	hide() {
+		this._showStore.clear();
+	}
+}
+
+class SimpleBrowserOverlayController {
+
+	private readonly _store = new DisposableStore();
+
+	private readonly _domNode = document.createElement('div');
+
+	constructor(
+		container: HTMLElement,
+		group: IEditorGroup,
+		@IInstantiationService instaService: IInstantiationService,
+		@IChatService chatService: IChatService,
+		@IChatEditingService chatEditingService: IChatEditingService,
+		@IInlineChatSessionService inlineChatService: IInlineChatSessionService
+	) {
+
+		this._domNode.classList.add('chat-editing-editor-overlay');
+		this._domNode.style.position = 'absolute';
+		this._domNode.style.bottom = `5px`;
+		this._domNode.style.right = `5px`;
+		this._domNode.style.zIndex = `100`;
+
+		const widget = instaService.createInstance(SimpleBrowserOverlayWidget, group, container);
+		this._domNode.appendChild(widget.getDomNode());
+		this._store.add(toDisposable(() => this._domNode.remove()));
+		this._store.add(widget);
+
+		const show = () => {
+			if (!container.contains(this._domNode)) {
+				container.appendChild(this._domNode);
+			}
+		};
+
+		const hide = () => {
+			if (container.contains(this._domNode)) {
+				widget.hide();
+				this._domNode.remove();
+			}
+		};
+
+		const activeEditorSignal = observableSignalFromEvent(this, Event.any(group.onDidActiveEditorChange, group.onDidModelChange));
+
+		const activeUriObs = derivedOpts({ equalsFn: isEqual }, r => {
+
+			activeEditorSignal.read(r); // signal
+
+			const editor = group.activeEditorPane;
+			if (editor?.input.editorId === 'mainThreadWebview-simpleBrowser.view') {
+				const uri = EditorResourceAccessor.getOriginalUri(editor?.input, { supportSideBySide: SideBySideEditor.PRIMARY });
+				return uri;
+			}
+			return undefined;
+		});
+
+		this._store.add(autorun(r => {
+
+			const data = activeUriObs.read(r);
+
+			if (!data) {
+				hide();
+				return;
+			}
+
+			// widget.show();
+			show();
+
+
+		}));
+	}
+
+	dispose(): void {
+		this._store.dispose();
+	}
+}
+
+export class SimpleBrowserOverlay implements IWorkbenchContribution {
```

Comment:
Not sure where are going this but if editor overlays become "a thing" then we should make this more proper. We should have some kind of service where you register for an editor type and it calls you back to populate your overlay